test/test_node.go

```diff
@@ -266,46 +268,43 @@ func isReachable(wg *sync.WaitGroup, t *testing.T, host ContainerPort, ch chan<-
 	ch <- connectionAttempt(t, host)
 }
 
-func (hosts Hosts) WaitForAllToStart(t *testing.T, timeout int) {
+func (hosts Hosts) WaitForAllToStart(t *testing.T, timeout int) error {
 	if len(hosts) == 0 {
-		return
+		return nil
 	}
 	for seconds := 1; seconds <= timeout; seconds++ {
 		var wg sync.WaitGroup
 
 		results := make(chan bool, len(hosts))
-		for i := 0; i < len(hosts); i++ {
-			host := hosts[i]
-			wg.Add(1)
+		wg.Add(len(hosts))
+		for _, host := range hosts {
 			go isReachable(&wg, t, host, results)
 		}
+		wg.Wait()
 
-		go func() {
-			wg.Wait()
-			close(results)
-		}()
+		close(results)
 
 		foundUnreachable := false
 
 		for reachable := range results {
 			if !reachable {
-				t.Logf("A signer node is not reachable")
+				t.Logf("A host is not reachable")
 				foundUnreachable = true
```

Comment:
This might be cleaner as a labeled continue statement. https://go.dev/ref/spec#Continue_statements